test/unit/PackagePluginSpec.hs

```diff
@@ -200,8 +200,6 @@ packageSpec = do
                   , "    - -threaded\n"
                   , "    - -rtsopts\n"
                   , "    - -with-rtsopts=-N\n"
-                  , "    dependencies:\n"
-                  , "    - asdf\n"
```

Comment:
This is very strange. It looks like this test-case should never have succeeded, but for some reason it did.
However, now we remove this, since it should actually not append the dependecies to both, global dependencies and dependencies for a stanza

Reply:
Why should it have failed?

Reply:
Because the dependency of `zlib` should have only been added to either the top-level `dependencies` or to all stanzas, the file for which we add the dependency, belongs to.

So, the test-case, in my definition, is just wrong. Now it is correct. But for some reason, this test-case succeeded before this PR and I dont know why. Or why it fails now.

Reply:
I didn't think the dependencies stanza did anything currently?

Reply:
it does in package.yaml